### PR TITLE
[super mega ultra WIP turbo II] Debian packaging via dh_virtualenv

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

File: build_debian.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (richvdh) [inline build_debian.sh]: I'm not super keen on this being in the root. It looks like a docker thing anyway - can it go in the docker dir?

(and can it have a comment saying what it does?)

File: .gitignore [hunk 2]
```diff
@@ -57,3 +58,6 @@ env/
 
 .vscode/
 .ropeproject/
+
+debs/
```
Comment (richvdh) [inline .gitignore]:
```suggestion
/debs/
```

File: .gitignore [hunk 3]
```diff
@@ -19,6 +19,7 @@ homeserver*.log
 homeserver*.log.*
 homeserver*.pid
 homeserver*.yaml
+!debian/homeserver.yaml
```
Comment (richvdh) [inline .gitignore]:
```suggestion
!/debian/homeserver.yaml
```

File: debian/.gitignore [hunk 4]
```diff
@@ -0,0 +1,9 @@
+matrix-synapse.*.debhelper
```
Comment (richvdh) [inline debian/.gitignore]: these should all have `/` prefixes I think

File: debian/.gitignore [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+matrix-synapse.debhelper.log
+matrix-synapse.substvars
+matrix-synapse/
+matrix-synapse-*/
+files
```
Comment (richvdh) [inline debian/.gitignore]: where does this come from?

File: debian/.gitignore [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+debhelper-build-stamp
+source/
```
Comment (richvdh) [inline debian/.gitignore]: shouldn't this be in vcs?

File: debian/changelog [hunk 7]
```diff
@@ -0,0 +1,5 @@
+matrix-synapse (0.33.9) bionic; urgency=low
```
Comment (richvdh) [inline debian/changelog]: can you give it a different name to distinguish it from the py2 packaging please.

File: debian/changelog [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+
+  * matrix-synapse: Initial debian packaging
```
Comment (richvdh) [inline debian/changelog]: not really...

File: debian/changelog [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+
+ -- Synapse Developers <synapse@matrix.org> Day, 01 Jan 2014 12:34:56 +0000
```
Comment (richvdh) [inline debian/changelog]: /me checks calendar

File: debian/control [hunk 10]
```diff
@@ -0,0 +1,20 @@
+Source: matrix-synapse
+Section: contrib/python
+Priority: extra
+Maintainer: Synapse Developers <synapse@matrix.org>
+Build-Depends: debhelper (>= 9), python3, python3-dev, libffi-dev, libssl-dev, libjpeg-dev, libxslt1-dev, dh-virtualenv (>= 0.10), tar, lsb-release
```
Comment (richvdh) [inline debian/control]: please can you break up the dependencies lines into one package per line, sorted alphabetically

File: debian/control [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+Standards-Version: 3.9.5
+Homepage: https://github.com/matrix-org/synapse
+
+Package: matrix-synapse
+Architecture: any
+Pre-Depends: dpkg (>= 1.16.1), python3, ${misc:Pre-Depends}
```
Comment (richvdh) [inline debian/control]: why do we pre-depend on python3?

File: debian/cookiecutter.json [hunk 12]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (richvdh) [inline debian/cookiecutter.json]: what's this for?

File: debian/.gitignore (hunk 6 above)
Comment (hawkowl) [reply debian/.gitignore]: no, because it's loaded from the mounted dir and is purely temporary while building.

Comment (richvdh) [reply debian/.gitignore]: what do you mean by "the mounted dir"?

Comment (hawkowl) [reply debian/.gitignore]: the synapse repo is mounted into the docker image, and as part of building, it copies it from the mounted source dir into the debian/ dir to work on as a copy.

File: debian/changelog (hunk 7 above)
Comment (hawkowl) [reply debian/changelog]: done

**Comment by hawkowl** [comment]
Closed in favour of https://github.com/matrix-org/synapse/pull/4285

**Pull request closed without merging**